test/metabase/driver/sql/query_processor_test.clj

```diff
@@ -254,22 +255,26 @@
     (binding [hx/*honey-sql-version* 2]
       (with-redefs [driver/db-start-of-week   (constantly :monday)
                     setting/get-value-of-type (constantly :sunday)]
-        (is (= [:dateadd
-                (hx/literal "day")
-                (hx/with-database-type-info [:cast [:inline -1] [:raw "long"]] "long")
-                (hx/with-database-type-info
-                  [:cast
-                   [:week [:dateadd (hx/literal "day")
-                           (hx/with-database-type-info [:cast [:inline 1] [:raw "long"]] "long")
-                           (hx/with-database-type-info [:cast :created_at [:raw "datetime"]] "datetime")]]
-                   [:raw "datetime"]]
-                  "datetime")]
-               (sql.qp/adjust-start-of-week :h2 (partial hx/call :week) :created_at)))))
+        (is (= (-> (h2x/with-database-type-info
+                    [:dateadd
+                     (h2x/literal "day")
+                     [:inline -1]
+                     (h2x/with-database-type-info
+                      [:cast
+                       [:week (h2x/with-database-type-info
+                               [:dateadd (h2x/literal "day")
+                                [:inline 1]
+                                (h2x/with-database-type-info [:cast :created_at [:raw "datetime"]] "datetime")]
+                               "datetime")]
+                       [:raw "datetime"]]
+                      "datetime")]
+                    "datetime"))
+               (sql.qp/adjust-start-of-week :h2 (fn [x] [:week x]) :created_at)))))
```

Comment:
Optional style/taste thing: I find `(f some-big-long-expression short-value)` hard to read, and
`(-> some-big-long-expression (f short-value))` easier.

Reply:
fixed